### ISSUE TITLE
EMA Model Save Fails Due to CPU Memory Exhaustion

### ISSUE THREAD

**Issue by minghuix98** [ByteDance-Seed/Bagel#204]
I can save the model ckpt after fine-tuning it on 2 A100 GPUs with  --cpu_offload True and --sharding_strategy FULL_SHARD.
But the EMA model can't be saved correctly (due to CPU memory exhaustion, ~433 GB) even I consider:
with FSDP.state_dict_type(
        ema_model,
        StateDictType.SHARDED_STATE_DICT,
        ShardedStateDictConfig(offload_to_cpu=True),
    ):
...

Have anyone encounter this?
I'd appreciate your help!

**Comment by minghuix98** [comment]
I think that it's caused by '.state_dict()' method, which is hard to address.
I tried torch.distributed.checkpoint but face another problem: Some typeerror pickle ...
